From 23a9622196b98234ca69f473ca3ef11639d35b2d Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Mon, 14 Oct 2013 15:19:20 -0400
Subject: [PATCH 0212/1074] ENGR00283508-3 keyboard: imx: support pin sleep
 mode in suspend

When system suspend, need to switch keyboard pins to sleep mode,
this is to save IO power. Using pinctrl mode switch framework
to make it happen.

Signed-off-by: Anson Huang <b20788@freescale.com>
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 drivers/input/keyboard/imx_keypad.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/drivers/input/keyboard/imx_keypad.c b/drivers/input/keyboard/imx_keypad.c
index cbf4f80..047a135 100644
--- a/drivers/input/keyboard/imx_keypad.c
+++ b/drivers/input/keyboard/imx_keypad.c
@@ -1,6 +1,7 @@
 /*
  * Driver for the IMX keypad port.
  * Copyright (C) 2009 Alberto Panizzo <maramaopercheseimorto@gmail.com>
+ * Copyright (C) 2013 Freescale Semiconductor, Inc.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
@@ -548,6 +549,8 @@ static int imx_kbd_suspend(struct device *dev)
 
 	if (device_may_wakeup(&pdev->dev))
 		enable_irq_wake(kbd->irq);
+	else
+		pinctrl_pm_select_sleep_state(dev);
 
 	return 0;
 }
@@ -561,6 +564,8 @@ static int imx_kbd_resume(struct device *dev)
 
 	if (device_may_wakeup(&pdev->dev))
 		disable_irq_wake(kbd->irq);
+	else
+		pinctrl_pm_select_default_state(dev);
 
 	mutex_lock(&input_dev->mutex);
 
-- 
1.7.5.4

